tasks/upgrades.ts

```diff
@@ -0,0 +1,317 @@
+import { task, types } from "hardhat/config";
+import { MarginEngine, VAMM } from "../typechain";
+import { BigNumber, ethers } from "ethers";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import mustache from "mustache";
+import * as fs from "fs";
+import path from "path";
+import { getRateOracleByNameOrAddress } from "./helpers";
+
+const _ERC1967_IMPLEMENTATION_SLOT =
+  "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc";
+interface UpgradeTemplateData {
+  proxyUpgrades: { proxyAddress: string; newImplementation: string }[];
+  rateOracleUpdates: {
+    marginEngineAddress: string;
+    vammAddress: string;
+    rateOracleAddress: string;
+    lookbackWindowInSeconds: BigNumber;
+  }[];
+}
+
+const proxiedContractTypes = [
+  "VAMM",
+  "MarginEngine",
+  // "Periphery",
```

Comment:
@CyclopsRex @ioanaroceanu2000 periphery should be added to the list following the upgradability changes introduced by Ioana